src/metkit/mars/TypeToByList.h

```diff
@@ -9,39 +9,137 @@
  */
 
 /// @file   TypeToByList.h
-/// @author Baudouin Raoult
-/// @author Tiago Quintino
-/// @date   April 2016
+/// @author Emanuele Danovaro
+/// @date   March 2025
 
-#ifndef metkit_TypeToByList_H
-#define metkit_TypeToByList_H
+#pragma once
 
-#include "metkit/mars/TypeInteger.h"
+#include "eckit/exception/Exceptions.h"
+#include "eckit/utils/StringTools.h"
+#include "eckit/utils/Translator.h"
 
-namespace metkit {
-namespace mars {
+#include "metkit/mars/Type.h"
+
+namespace metkit::mars {
+
+class MarsExpandContext;
 
 //----------------------------------------------------------------------------------------------------------------------
+template <typename EL, typename BY>
+class TypeToByList : public ITypeToByList {
 
-class TypeToByList : public TypeInteger {
+private:  // members
+
+    const Type* type_;
+    const std::string by_;
 
 public:  // methods
 
-    TypeToByList(const std::string& name, const eckit::Value& settings);
+    TypeToByList(const Type* type, const eckit::Value& settings) :
+        type_(type), by_(settings.contains("by") ? settings["by"] : "1") {}
+
+    virtual ~TypeToByList() = default;
+
+    void expandRanges(const MarsExpandContext& ctx, std::vector<std::string>& values) const override {
+
+        static eckit::Translator<std::string, EL> s2el;
+        static eckit::Translator<std::string, BY> s2by;
+        static eckit::Translator<EL, std::string> el2s;
+
+        std::vector<std::string> newval;
+
+        for (size_t i = 0; i < values.size(); ++i) {
 
-    virtual ~TypeToByList() override;
+            const std::string& s = values[i];
 
-private:  // methods
+            if (eckit::StringTools::lower(s) == "to" || eckit::StringTools::lower(s) == "t0") {
```

Comment:
wait, what. Why is t0 permitted. That is weird. Can you confirm that to me in chat.